presto-hive/src/main/java/io/prestosql/plugin/hive/metastore/glue/GlueMetastoreModule.java

```diff
@@ -57,4 +65,14 @@ protected void setup(Binder binder)
                     .as(generator -> generator.generatedNameOf(GlueHiveMetastore.class));
         }
     }
+
+    @Provides
+    @Singleton
+    @ForGlueHiveMetastore
+    public Executor createCachingHiveMetastoreExecutor(HiveCatalogName catalogName, GlueHiveMetastoreConfig hiveConfig)
+    {
+        return new BoundedExecutor(
+            newCachedThreadPool(daemonThreadsNamed("hive-glue-%s")),
+            hiveConfig.getMaxGlueGetPartitionThreads());
```

Comment:
When `hiveConfig.getMaxGlueGetPartitionThreads()` we could return `directExecutor()` here.

Reply:
Sure